### PR TITLE
Not download firmware if SOF works in legacy HD-A mode

### PR DESCRIPTION
**Pull request by mengdonglin**
This PR is to skip firmware download if SOF is working in legacy HD-A mode. User can use Kconfig to decide whether to put SOF in legacy mode. So the patches are okay for topic/hdac branch.

### CODE REVIEW

File: sound/soc/sof/intel/Kconfig [hunk 1]
```diff
@@ -104,4 +104,13 @@ config SND_SOC_SOF_HDA
           Say Y if you want to enble HDA links with SOF.
           If unsure select "N".
 
+config SND_SOC_SOF_BYPASS_DSP
```
Comment (lgirdwood) [inline sound/soc/sof/intel/Kconfig]: Change this to include the word FORCE as we are forcing HDA legacy mode.

File: sound/soc/sof/intel/Kconfig [hunk 2]
```diff
@@ -104,4 +104,13 @@ config SND_SOC_SOF_HDA
...
+	tristate "SOF bypass DSP"
+	depends on SND_SOC_SOF_HDA
+	help
+          This adds support for SOF in legacy HD-Audio mode which bypasses
```
Comment (lgirdwood) [inline sound/soc/sof/intel/Kconfig:111]: This forces the driver to use HDA legacy mode....

File: sound/soc/sof/sof-acpi-dev.c [hunk 3]
```diff
@@ -134,6 +134,7 @@ static struct platform_device *
 	return pdev;
 }
 
+#if !IS_ENABLED(CONFIG_SND_SOC_SOF_BYPASS_DSP)
```
Comment (lgirdwood) [inline sound/soc/sof/sof-acpi-dev.c:137]: Dont put the #ifdef here but put inside sof_create_audio_device()

File: sound/soc/sof/utils.c [hunk 4]
```diff
@@ -47,3 +48,21 @@ int sof_bes_setup(struct device *dev, struct snd_sof_dsp_ops *ops,
 }
 EXPORT_SYMBOL(sof_bes_setup);
 
+/* register sof audio device */
+int sof_create_audio_device(struct sof_platform_priv *priv)
```
Comment (lgirdwood) [inline sound/soc/sof/utils.c:52]: #fdefs should be in here for FORCE HDA mode.

**Comment by mengdonglin** [comment]
Close this PR with new implementation.

**Pull request closed without merging**